src/Cli.Tests/UtilsTests.cs

```diff
@@ -1,261 +1,269 @@
         {
-            TestHelper.SetupTestLoggerForCLI();
-        }
+            builder.AddConsole();
+        });
 
-        /// <summary>
-        /// Test to validate the REST Path constructed from the input entered using
-        /// --rest option  
-        /// </summary>
-        /// <param name="restRoute">REST Route input from the --rest option</param>
-        /// <param name="expectedRestPath">Expected REST path to be constructed</param>
-        [DataTestMethod]
-        [DataRow(null, null, DisplayName = "No Rest Path definition")]
-        [DataRow("true", true, DisplayName = "REST enabled for the entity")]
-        [DataRow("false", false, DisplayName = "REST disabled for the entity")]
-        [DataRow("customPath", "/customPath", DisplayName = "Custom REST path defined for the entity")]
-        public void TestContructRestPathDetails(string? restRoute, object? expectedRestPath)
-        {
-            object? actualRestPathDetails = ConstructRestPathDetails(restRoute);
-            Assert.AreEqual(expectedRestPath, actualRestPathDetails);
-        }
+        SetLoggerForCliConfigGenerator(loggerFactory.CreateLogger<ConfigGenerator>());
+        SetCliUtilsLogger(loggerFactory.CreateLogger<Utils>());
+    }
 
-        /// <summary>
-        /// Test to validate the GraphQL Type constructed from the input entered using
-        /// --graphql option
-        /// </summary>
-        /// <param name="graphQLType">GraphQL Type input from --graphql option</param>
-        /// <param name="expectedGraphQLType">Expected GraphQL Type to be constructed</param>
-        [DataTestMethod]
-        [DataRow(null, null, false, DisplayName = "No GraphQL Type definition")]
-        [DataRow("true", true, false, DisplayName = "GraphQL enabled for the entity")]
-        [DataRow("false", false, false, DisplayName = "GraphQL disabled for the entity")]
-        [DataRow("book", null, true, DisplayName = "Custom GraphQL type - Singular value defined")]
-        [DataRow("book:books", null, true, DisplayName = "Custom GraphQL type - Singular and Plural values defined")]
-        public void TestConstructGraphQLTypeDetails(string? graphQLType, object? expectedGraphQLType, bool isSingularPluralType)
-        {
-            object? actualGraphQLType = ConstructGraphQLTypeDetails(graphQLType);
-            if (!isSingularPluralType)
-            {
-                Assert.AreEqual(expectedGraphQLType, actualGraphQLType);
-            }
-            else
-            {
-                SingularPlural expectedType = new(Singular: "book", Plural: "books");
-                Assert.AreEqual(expectedType, actualGraphQLType);
-            }
+    [TestMethod]
+    public void ConstructRestOptionsForCosmosDbNoSQLIgnoresOtherParamsAndDisables()
+    {
+        EntityRestOptions options = ConstructRestOptions("true", Array.Empty<SupportedHttpVerb>(), true);
+        Assert.IsFalse(options.Enabled);
+    }
 
-        }
+    [TestMethod]
```

Comment:
Why not use DataTestMethod, to save lines of code?

Reply:
there are multiple methods which has exact same lines of code only change is in the input and output. Is there any specific reason we chose to move out of DataRow?

Reply:
It use to be a `DataTestMethod` but I split it apart (same for the GraphQL one further down) because the refactor away from having `object?` returned from `ConstructRestOptions` to using the strong type would result in a lot of different branch logic in the test itself (which we previously did).

Best practice is to avoid branch logic within tests because that adds logic to the tests which should be tested as well.